src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -94,8 +94,8 @@ protected override async Task LoadComponentsAsync(CancellationToken cancellation
             _packageInstallerService = Workspace.Services.GetService<IPackageInstallerService>() as PackageInstallerService;
             _symbolSearchService = Workspace.Services.GetService<ISymbolSearchService>() as VisualStudioSymbolSearchService;
 
-            _packageInstallerService?.Connect(this.RoslynLanguageName);
-            _symbolSearchService?.Connect(this.RoslynLanguageName);
+            _packageInstallerService?.Connect(this.RoslynLanguageName, this.DisposalToken);
+            _symbolSearchService?.Connect(this.RoslynLanguageName, this.DisposalToken);
```

Comment:
cancellation token handling was wonky.  we effectively want this stuff to run and do async work until teh package closes.  this ties these services to that concept.  Note: another way to do this might be to have the package itself export some sort of IGlobalShutdown interface that exposes this token so we can more cleanly import it.  However, given that the package directly communicates with tehse two services, having it simply inform them of the disposable token seems fine.

Reply:
@jasonmalinowski .  I can wait on that.  Or i can do this, and then just update this code once that PR goes in.  Do you have a preference?

Reply:
I don't have a preference either way. I see @sharwell has moved that one back to draft -- I'll let you both decide which order you want to do it in. If @sharwell's goes in first then you can use it, otherwise he can fix this up.

Reply:
I'll get mine updated today. I was delayed by the 16.7 keep alive issue.

Reply:
that works for me.  Regardless of teh order, i'll make sure to use the new service.  It's exactly what i wanted, i just didn't want to go through the effort of creating it myself in this PR.

Reply:
this has been addressed.